### DISCUSSION TITLE
Android screen - How to use the Kotlin top bar (search bar) area as regular text?

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#267] [Q&A] [answered]
I've noted that in most of my Android app pages - there is a significant area left at the top of every page... I guess it's the top search bar area in Kotlin world.  Any ideas on how I use this space?  I tried the SwiftUI .ignoreSafeArea on the VStack of my page.  This resulted in all the screen being used... way too much.  Is there a recommendation on Skip World - for using the "realistic" full screen?
<img width="695" alt="Screenshot 2024-10-16 at 9 35 32 PM" src="https://github.com/user-attachments/assets/db8d23aa-432c-402b-bcca-8e6b1761d158">
<img width="281" alt="Screenshot 2024-10-16 at 9 35 39 PM" src="https://github.com/user-attachments/assets/1ed35f01-f547-4b1d-a41f-7dfd1bb3c355">
<img width="284" alt="Screenshot 2024-10-16 at 9 39 21 PM" src="https://github.com/user-attachments/assets/e343fc7d-39b5-4b11-b353-8c1d51157202">

**Comment by aabewhite** [comment, marked as answer]
That's the navigation bar. You can hide it: https://www.hackingwithswift.com/quick-start/swiftui/how-to-hide-the-tab-bar-navigation-bar-or-other-toolbars

**Comment by davidakoontz** [comment]
Thank you just what I needed. David On Oct 16, 2024, at 10:22 PM, Abe White ***@***.***> wrote:﻿
That's the navigation bar. You can hide it: https://www.hackingwithswift.com/quick-start/swiftui/how-to-hide-the-tab-bar-navigation-bar-or-other-toolbars

**Comment by davidakoontz** [comment]
Thank you Abe, that got me pointed in the right direction.  Now I'm in a dilemma... the hiding of the Nav bar in SwiftUI removes the automatic "< Back" button that detial pages naturally get... so navigation is broken.

I've tried:
toolbar(.hidden)

which closes up all that whitespace and uses the full screen; but I lose the "< Back" button on the detail page with no Nav return.

I've tried:
.navigationBarBackButtonHidden(true)
.toolbar { Button {} }

![Screenshot 2024-10-17 at 7 15 26 PM](https://github.com/user-attachments/assets/33484cc6-97db-4d3f-8b39-1b27d406f917)

results look fine in iOS but not Android... still a lot of WhiteSpace before content.

Putting this out here ... for help if anyone has thoughts???

**Reply by aabewhite** [reply]
As soon as you put a button on .toolbar you're bringing a top bar back into the picture.

You can keep the top bar but make it smaller using .navigationBarTitleDisplayMode(.inline)

Or you can continue to use .hidden and place a custom back button onscreen yourself using normal SwiftUI layout mechanisms. If you choose that solution, the way to make it perform the normal "back" functionality on tap is to use the "dismiss" action like so:

```
struct MyView: View {
    @Environment(\.dismiss) var dismiss
    
    var body: some View {
        ...
        Button("Back") { dismiss() }
        ...   
    }
}
```

**Reply by davidakoontz** [reply]
Oh you are the BEST!  The trick for me is:
keep the top bar but make it smaller using .navigationBarTitleDisplayMode(.inline)

It keeps the "< Back" button functional and removes most of the white space that looked too weird.  SHIP IT!
<img width="281" alt="Screenshot 2024-10-18 at 2 13 48 PM" src="https://github.com/user-attachments/assets/1d58c512-7e3f-4332-b7e2-3fcfaa437a35">

**Reply by davidakoontz** [reply]
still lot's of white space (header) on the page ... that I'd love to reclaim - but got to have the back button functionality also... and don't wont to go into designing every detail on the page - this App is an MVP... we've got important things to learn (none of which are Kotlin screen layout).
